company/forms.py

```diff
@@ -37,6 +40,33 @@ class Meta:
         ]
 
 
+class MoveProductsForm(SingleButtonMixin, forms.Form):
+    new_company = forms.ModelChoiceField(
+        queryset=Company.objects.all(),
+        widget=Select(),
+        label="Wybierz firmę do której produkty mają być przeniesione",
+    )
+
+    def save(self, pk):
+        failed = []
+        success = []
+        new_company = self.cleaned_data['new_company']
+        if new_company.pk == pk:
+            failed.append("the_same_company")
+            return success, failed
+        products = Product.objects.filter(company=pk)
+        count = products.count()
+        if count == 0:
+            failed.append("no_products")
```

Comment:
To jest walidacja konkretnego pola, więc powinno być w metodzie `clean_source_company`